src/ert/cli/main.py

```diff
@@ -104,7 +104,43 @@ def run_cli(args: Namespace, plugin_manager: ErtPluginManager | None = None) ->
         # This is within the range for ephemeral ports as defined by
         # most unix flavors https://en.wikipedia.org/wiki/Ephemeral_port
         args.port_range = range(49152, 51819)
+    if (
+        ert_config.analysis_config.design_matrix is not None
+        and ert_config.analysis_config.design_matrix.active_realizations is not None
+    ):
+        dm_num_realizations = (
+            ert_config.analysis_config.design_matrix.active_realizations
+        )
+        config_num_realizations = ert_config.runpath_config.num_realizations
 
+        if (
+            hasattr(args, "realizations")
+            and args.realizations is None
+            and len(dm_num_realizations) != ert_config.runpath_config.num_realizations
+        ):
+            if (
+                len(ert_config.analysis_config.design_matrix.active_realizations)
```

Comment:
This seems to be incorrect. We have discussed that num_realizations should always take the precedents, or?